src/generic_ui/polymer/tip.ts

```diff
@@ -0,0 +1,64 @@
+Polymer({
+  close: function() {
+    this.fire('closed');
+    this.remove();
+  },
+  doReposition: function() {
+    var prev = this.previousSibling;
+    if (prev.offsetLeft === 0 && prev.offsetTop === 0 && prev.offsetWidth === 0
+        && prev.offsetHeight === 0) {
+      /* if the DOM is not set up yet, keep checking every half second */
```

Comment:
I'm not completely clear on the terminology for that comment, it's meant to handle when everything is hidden and elements do not have positions even though the DOM is technically "set up".

I was actually wanting to set things up that way (in my personal ideal world, the entire app would be set up that way), however, I ran into trouble figuring out where we would want to be creating the element that had the ability to work with the root (ideally we'd do it when you finish signing in, but that is taken care of in `network.ts` where there is definitely not the ability to access the elements in `root.html`.  If you have any ideas for how to make that work, I'd absolutely love to do it.